cpp/daal/src/algorithms/dtrees/forest/classification/df_classification_train_dense_default_impl.i

```diff
@@ -382,8 +405,8 @@ bool UnorderedRespHelper<algorithmFPType, cpu>::findBestSplitOrderedFeature(cons
         split.iStart       = 0;
         DAAL_ASSERT(split.nLeft >= nMinSplitPart);
         DAAL_ASSERT((n - split.nLeft) >= nMinSplitPart);
-        DAAL_ASSERT(split.nLeft >= minWeightLeaf);
-        DAAL_ASSERT((n - split.nLeft) >= minWeightLeaf);
+        DAAL_ASSERT(split.nLeftWeight >= minWeightLeaf);
+        DAAL_ASSERT((split.nTotalWeight - split.nLeftWeight) >= minWeightLeaf);
```

Comment:
Very questionable checks. Double-check that they should be in the term of floating point calculations